echopype/convert/parse_base.py

```diff
@@ -639,13 +639,15 @@ def pad_shorter_ping(data_list) -> np.ndarray:
             # Create output array from mask
             out_array = np.full(mask.shape, np.nan)
 
+            # Concatenate short pings
+            concat_short_pings = np.concatenate(data_list).reshape(-1)  # reshape in case data > 1D
+
             # Take care of problem of np.nan being implicitly "real"
-            arr_dtype = data_list[0].dtype
-            if np.issubdtype(arr_dtype, np.complex_):
-                out_array = out_array.astype(arr_dtype)
+            if concat_short_pings.dtype == np.complex64:
+                out_array = out_array.astype(np.complex64)
```

Comment:
I think my question was whether the original `astype(arr_dtype)` is needed to make sure dtypes other than `np.complex64` would be set correctly, but I am unsure.

Reply:
but it will only be called if the dtype is equal to complex64. i think the original intention of the astype(arr_dtype) was that (presumably) the types that are subdtype of complex_ could be complex64/128/_ etc. but since i am just checking that the dtype is equal to complex64, i dont think we will run into any other case like that

Reply:
Oh I think you're right! Sorry I completely misread the if statement.